sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -178,6 +200,9 @@ object ScalaNativePluginInternal {
     }
   )
 
+  private def writeConfig(file: File, config: Any*): Unit =
+    IO.write(file, config.##.toString)
```

Comment:
Something like sbt's `sbt.Hash(configString)` would be more robust. Scala's hash is only 32 bits, compared to 160 bits of SHA-1 of `sbt.Hash`.